src/cobalt.rs

```diff
@@ -162,9 +172,43 @@ fn generate_pages(posts: Vec<Document>, documents: Vec<Document>, context: &Cont
     trace!("Generating other documents");
     for mut doc in documents {
         trace!("Generating {}", doc.url_path);
-        generate_doc(&posts_data, &mut doc, context)?;
+        if doc.front.pagination.is_some() {
+            trace!("It's an index page.");
+            let paginators = pagination::generate_paginators(&mut doc, &posts_data)?;
+            // page 1 uses frontmatter.permalink instead of paginator.permalink
+            let mut paginators = paginators.into_iter();
+            let paginator = paginators
+                .next()
+                .expect("We detected pagination enabled but we have no paginator");
+            generate_doc(
+                &mut doc,
+                context,
+                (
+                    "paginator".into(),
+                    liquid::value::Value::Object(paginator.into()),
+                ),
+            )?;
+            for paginator in paginators {
+                let mut doc_page = doc.clone();
+                doc_page.file_path =
+                    permalink::format_url_as_file(paginator.index_permalink.clone());
```

Comment:
without it:
```
error[E0382]: use of partially moved value: `paginator`
   --> src\cobalt.rs:200:54
    |
194 |                     permalink::format_url_as_file(paginator.index_permalink);
    |                                                   ------------------------- value moved here
...
200 |                         liquid::value::Value::Object(paginator.into()),
    |                                                      ^^^^^^^^^ value used here after move
    |
    = note: move occurs because `paginator.index_permalink` has type `std::string::String`, which does not implement the `Copy` trait
```